prefect_census/syncs.py

```diff
@@ -0,0 +1,220 @@
+"""Module containing tasks and flows for interacting with Census syncs."""
+from httpx import HTTPStatusError
+from prefect import flow, task
+from prefect.logging import get_run_logger
+
+from prefect_census.credentials import CensusCredentials
+from prefect_census.runs import (
+    CensusSyncRunCancelled,
+    CensusSyncRunFailed,
+    CensusSyncRunStatus,
+    wait_census_sync_completion,
+)
+from prefect_census.utils import extract_user_message
+
+
+class CensusSyncTriggerFailed(RuntimeError):
+    """Used to indicate sync triggered."""
+
+    pass
+
+
+@task(
+    name="Trigger Census sync run",
+    description="Triggers a Census sync run for the sync " "with the given sync_id.",
+    retries=3,
+    retry_delay_seconds=10,
+)
+async def trigger_census_sync(credentials: CensusCredentials, sync_id: int) -> dict:
+    """
+    A task to trigger a Census sync run.
+
+    Args:
+        credentials: Credentials for authenticating with Census.
+        sync_id: The ID of the sync to trigger.
+
+    Returns:
+        The run data returned from the Census API.
+
+    Examples:
+        Trigger a Census sync run:
+        ```python
+        from prefect import flow
+
+        from prefect_census.credentials import CensusCredentials
+        from prefect_census.syncs import trigger_census_sync
+
+        @flow
+        def trigger_census_sync_flow():
+            credentials = CensusCredentials(api_key="my_api_key")
+            trigger_census_sync(credentials=credentials, sync_id=42)
+
+        trigger_census_sync_flow()
+        ```
+    """  # noqa
```

Comment:
Do we need `#noqa` here?
```suggestion
    """
```